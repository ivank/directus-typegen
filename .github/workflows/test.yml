name: Tests

on:
  pull_request:
    branches: [main, master]
  push:
    branches: [main, master]

jobs:
  test:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v4

      - name: Setup Bun
        uses: oven-sh/setup-bun@v1
        with:
          bun-version: latest

      - name: Install dependencies
        run: bun install

      - name: Format code
        run: bun format

      - name: Format code
        run: bun build

      - name: Run tests
        run: bun test
