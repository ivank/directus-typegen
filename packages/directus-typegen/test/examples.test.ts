import { describe, test, expect } from 'bun:test';
import { spawn } from 'bun';
import { readdirSync } from 'fs';
import { join } from 'path';

const examplesDir = join(import.meta.dir, '../examples');

async function execute(args: string[], stdin?: Uint8Array) {
  const process = spawn(args, { stdin, stdout: 'pipe', stderr: 'pipe' });

  const [stdout, stderr, exitCode] = await Promise.all([
    new Response(process.stdout).text(),
    new Response(process.stderr).text(),
    process.exited,
  ]);

  return { stdout, stderr, exitCode };
}

describe('Example', () => {
  const exampleFiles = readdirSync(examplesDir).filter((file) => file.endsWith('.ts'));

  for (const exampleFile of exampleFiles) {
    test(`Should successfully run ${exampleFile}`, async () => {
      const examplePath = join(examplesDir, exampleFile);
      const result = await execute(['bun', examplePath]);

      expect(result).toMatchSnapshot();
    });
  }
});
