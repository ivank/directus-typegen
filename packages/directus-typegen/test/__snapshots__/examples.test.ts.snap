// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Should successfully run schema.ts 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": 
"{
  collections: Map(1) {
    "website": {
      isSingleton: true,
      fields: Map(3) {
        "id": [Object ...],
        "title": [Object ...],
        "content": [Object ...],
      },
    },
  },
  builtInCollections: Map {},
  junctions: Map {},
}
"
,
}
`;

exports[`Example Should successfully run simple.ts 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": 
"export interface website {
    id: string;
    title: string;
    content?: string;
}
export interface DirectusSchema {
    website: website;
}

"
,
}
`;

exports[`Example Should successfully run validate.ts 1`] = `
{
  "exitCode": 0,
  "stderr": "",
  "stdout": 
"{
  collections: [
    {
      collection: "website",
      meta: [Object ...],
    }
  ],
  fields: [
    {
      field: "id",
      type: "uuid",
      collection: "website",
      schema: [Object ...],
      meta: [Object ...],
    }, {
      field: "title",
      type: "string",
      collection: "website",
      schema: [Object ...],
      meta: [Object ...],
    }, {
      field: "content",
      type: "text",
      collection: "website",
      schema: [Object ...],
      meta: [Object ...],
    }
  ],
  relations: [],
}
"
,
}
`;
